ui/raidboss/raidboss.js

```diff
@@ -40,4 +40,20 @@ gTimelineController = new TimelineController(Options, new TimelineUI(Options));
 gPopupText = new PopupText(Options);
 // Connect the timelines to the popup text.
 gTimelineController.SetPopupTextInterface(new PopupTextGenerator(gPopupText));
-gPopupText.SetTimelineLoader(new TimelineLoader(gTimelineController));
\ No newline at end of file
+gPopupText.SetTimelineLoader(new TimelineLoader(gTimelineController));
+OverlayPluginApi.overlayMessage(OverlayPluginApi.overlayName, JSON.stringify({'getUserLocation': true}));
+document.addEventListener("onSendUserConfigLocation", function(e) {
```

Comment:
Can this code be encapsulated somehow? Maybe some resources/userconfig.js file can be included and then each module could just have a one liner of UserConfig.addConfigListener('raidboss') or something.

Also, it looks like the location config is how you're getting the "raidboss" part of `user\raidboss.js`? I think I would prefer if the location was a directory and not something like `C:\etc etc\cactbot\user\raidboss` where it ends with a partial file name.

Reply:
the location config is the returned string from the config file.  its the full path to the file

for CSS I'm then stripping the .js off and appending .css to load the custom user css.

So selecting say "raidboss-custom.js" in the raidboss's window config would load "raidboss-custom.js" and "raidboss-custom.css" but it can be named anything.  I could call it "purple-duck.js" and it would load that and "purple-duck.css"

Reply:
As for encapsulation, yes, that should be do-able, I'll have to double check what version of chromium is running to see if it has ES5 or ES6 support, if it has ES6 support I can do it with ES6 imports, if not ES5 imports should work, they just aren't as pretty as ES6 ones.

Reply:
Re: location config.  I think I'd still prefer to have it be a directory.  That lets files like user/raidboss-example.js and documentation tell users what files to create directly, and I can answer support questions by saying things like "put this in your cactbot/user/raidboss.js file".

Reply:
I'm used to working with Electron apps, which are node based, and C# chromium doesn't support modules like electron chromium does.  So, I wasn't able to do what I originally planned, but my fall-back to just putting it in the HTML works.